src/io_utils.rs

```diff
@@ -140,12 +142,78 @@ pub fn get_input_list(
 
 /// Checks if any input files are fastq
 pub fn any_fastq(files: &[InputFastx]) -> bool {
-    let mut fastq = false;
-    for file in files {
-        if file.2.is_some() {
-            fastq = true;
-            break;
+    files.iter().any(|file| file.2.is_some())
+}
+
+/// Counts number of fastq files
+pub fn count_fastq(files: &[InputFastx]) -> usize {
+    files.iter().filter(|file| file.2.is_some()).count()
+}
+
+/// Collects filepaths to the first 2 fastq files into a tuple
+pub fn get_2_fastq_path(files: &[InputFastx]) -> (String, String) {
+    let out: Vec<String> = files
+        .iter()
+        .filter(|file| file.2.is_some())
+        .take(2)
+        .map(|x| x.1.clone())
+        .collect();
+
+    if out.len().gt(&1) {
+        (out[0].clone(), out[1].clone())
+    } else {
+        panic!("Trying to get 2 fastq files from a vector with <2 elements");
+    }
+}
+
+/// Calculates minimum kmer cutoff depending on user provided argument
+pub fn kmer_min_cutoff(
+    v: &Option<ValidMinKmer>,
+    files: &[InputFastx],
+    rep_64: bool,
```

Comment:
Is it possible to make this generic over `IntT`? Then you can call:
```
CoverageHistogram::<IntT>::new(&fastq_fwd, &fastq_rev, *k, rc, verbose);
```
and remove the if below

Reply:
And you call then this with `u64`, `u128` specified in lib.rs